### ISSUE TITLE
Regression from version 3.2.11 to 3.2.12

### ISSUE THREAD

**Issue by AskMeAgain** [docker-java/docker-java#1798]
hi,

the following code works locally on version 3.2.11, but **not** on version 3.2.12.

Expected is when writing "echo test" to the docker container via exec, that the command is getting executed and the result is piped to stdout, but in version 3.2.12 nothing is happening. Switching to version 3.2.11 makes the code working again.

iam using these versions:

```
    docker_java_version = 3.2.11
    docker_java_version = 3.2.12

    implementation "com.github.docker-java:docker-java-transport-httpclient5:${docker_java_version}"
    implementation "com.github.docker-java:docker-java-core:${docker_java_version}"
```

```
@Slf4j
public class TestMain2 {
  public static void main(String[] args) throws InterruptedException, IOException {

    var dockerClient = getInstance();
    var container = dockerClient.createContainerCmd("alpine")
        .withCmd("tail", "-f","/dev/null")
        .exec();

    dockerClient.startContainerCmd(container.getId()).exec();

    var exec = dockerClient.execCreateCmd(container.getId())
        .withCmd("sh")
        .withTty(true)
        .withAttachStdout(true)
        .withAttachStderr(true)
        .withAttachStdin(true)
        .exec();

    var stdin = new PipedInputStream();
    var writeTo = new PipedOutputStream(stdin);

    dockerClient.execStartCmd(exec.getId())
        .withStdIn(stdin)
        .withTty(false)
        .exec((new ResultCallback.Adapter<>() {
          @Override
          public void onNext(Frame item) {
            System.out.println("Received: " + item.toString());
          }

          @Override
          public void onComplete() {
            System.out.println("Finished!");
          }
        }))
        .awaitStarted();

    Thread.sleep(3000);

    System.out.println("Echoing now!");
    writeTo.write("echo test\n".getBytes(StandardCharsets.UTF_8));
    Thread.sleep(3000);
  }


  private static DockerClient getInstance() {
    var config = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
    var httpClient = new ApacheDockerHttpClient.Builder()
        .dockerHost(config.getDockerHost())
        .sslConfig(config.getSSLConfig())
        .build();
    return DockerClientImpl.getInstance(config, httpClient);
  }
}
```

**Comment by derklaro** [comment]
Might be a duplicate of #1768 if you're running on Java 16 or later (does look like it)

**Comment by AskMeAgain** [comment]
Yes iam running java 17

**Comment by derklaro** [comment]
Should be fixed now, try with 3.2.13

**Comment by bsideup** [comment]
Duplicate of #1768

Also, consider using zerodep transport to avoid dependency version conficts in the future
